src/analysis/mesh/qgsmeshcalculator.h

```diff
@@ -62,6 +62,18 @@ class ANALYSIS_EXPORT QgsMeshCalculator
       MemoryError, //!< Error allocating memory for result
     };
 
+    /**
+     * Destination of the result of the calculation
+     *
+     * \since QGIS 3.16
+     */
+    enum Destination
```

Comment:
maybe something like this so it can be used also in various other places, and maybe defined in qgs_core somewhere?
```
enum MeshDatasetGroupStorage{
    PersistantMeshDatasetGroup,
    InMemoryMeshDatasetGroup,
    VirtualMeshDatasetGroup
}
```